### ISSUE TITLE
softmax wrong index in the GATConv and AGNNConv implementation? 

### ISSUE THREAD

**Issue by xwhan** [rusty1s/pytorch_geometric#224]
## ❓ Questions & Help

From my understanding, the default message flow is from the source to the target, in other words, from the edge_index[0] to edge_index[1]. In this sense the softmax should be used as `softmax(alpha, edge_index[1], num_nodes)` instead of `softmax(alpha, edge_index[0], num_nodes)` as in line 94 of GATConv.

**Comment by rusty1s** [comment]
Good catch! The softmax must be computed based on the message flow. Will fix ASAP.

**Comment by rusty1s** [comment]
Fixed in master! Thank you :)
